src/protocols/ethereum/libs/EthereumAdapter.ts

```diff
@@ -197,6 +199,26 @@ export class EthereumAdapter extends BaseProtocolAdapter {
     };
   }
 
+  override getAccountFromRawAccount(
+    rawAccount: IAccountRaw,
+    idx: number,
+    globalIdx: number,
+    seed: Uint8Array,
+  ): IAccount {
+    if (rawAccount.type !== ACCOUNT_TYPES.hdWallet) {
+      throw new Error(`Unsupported account type: ${rawAccount.type}`);
+    }
```

Comment:
I think this is bad to throw an error with a function called `getAccountFromRawAccount` where actually the app is supporting other account types than the hd wallet.